### ISSUE TITLE
GitVersionTask should not have a package dependency on Microsoft.NETFramework.ReferenceAssemblies

### ISSUE THREAD

**Issue by bording** [GitTools/GitVersion#2177]
As of the 5.2.0 release, the GitVersionTask package has a dependency on Microsoft.NETFramework.ReferenceAssemblies. This is incorrect because this package is for enabling  compiling .NET Framework assemblies on Linux. It has no runtime impact.

The reason for the unneeded dependency is because the package reference is not marked with `PrivateAssets="All"` in your [Directory.Build.props](https://github.com/GitTools/GitVersion/blob/master/src/Directory.Build.props#L42) file.

## Expected Behavior
The GitVersionTask package should have no dependencies.

## Actual Behavior
The GitVersionTask package does have a dependency:
![image](https://user-images.githubusercontent.com/753669/76635419-d5952600-651d-11ea-8e67-661a33b28f86.png)

## Possible Fix
Add `PrivateAssets="All"` to package reference in [Directory.Build.props](https://github.com/GitTools/GitVersion/blob/master/src/Directory.Build.props#L42) file.

**Comment by arturcic** [comment]
Good catch 👍

**Comment by arturcic** [comment]
Do you mind creating a PR with the fix? I will release a new version 5.2.3 afterwards

**Comment by bording** [comment]
I was actually in the process of creating a quick PR, but I've hit a snag with updating my fork:
```
 ! [remote rejected]   master -> master (refusing to allow an OAuth App to create or update workflow `.github/workflows/build.yml` without `workflow` scope)
```

Once I get that sorted, I'll get the PR created.
